controller/array_action/array_mediator_xiv.py

```diff
@@ -238,17 +230,21 @@ def _get_vol_by_wwn(self, volume_id):
 
     def delete_volume(self, volume_id):
         logger.info("Deleting volume with id : {0}".format(volume_id))
-        vol_name = self._get_vol_by_wwn(volume_id)
-
+        volume_name = self._get_vol_by_wwn(volume_id)
+        cli_snapshots = self.client.cmd.snapshot_list(vol=volume_name).as_list
+        if cli_snapshots:
+            raise controller_errors.ObjectIsStillInUseError(
+                id_or_name=[cli_snapshot.name for cli_snapshot in cli_snapshots],
```

Comment:
`id_or_name` doesn't sound plural. are you sure this is correct?

Reply:
generaly you pass there one volume but I have a list so I pass it all. I can't see the problem

Reply:
I still don't understand exactly the purpose of this.
could you explain the difference here comparing to svc?

Reply:
in svc we the return value of `id_or_name ` is the whole `unfinished_fcmaps`, same is here, we return the all the snapshots (I changed now to the all list just to be aligned with svc)

Reply:
done